### DISCUSSION TITLE
data from slice file using python: "fds_read_slice" (simulation time)

### DISCUSSION THREAD

**Discussion by jurvanoerle** [firemodels/fds#13621] [Q&A] [answered]
Hi,
I'm using the python script "fds_read_slice" to get data from slicefiles, stepping through the slices. 
(//github.com/firemodels/fds/tree/master/Utilities/Python/scripts)

That works ok, but I cant find/calculate the simulation time for each slicefile (like smokeview shows stepping through time)

regards,
Jur

**Comment by mcgratta** [comment]
I cannot find that script in the GitHub fds repository. Is this something you downloaded previously?

**Comment by jurvanoerle** [comment]
Hi,
I have used the script in https://github.com/firemodels/fds/blob/master/Utilities/Python/scripts/get_var_slice.py

I changed it a little bit (e.g. output to ascii file) and corrected a few typos, but now it works like a charm. 
At this moment I use the simulation period divided by the NRFRAMES (in the dump line) as the time for each timestep. I think that approach fails/is not accurate with a large number of dumpsteps and a short simulation period. 
I want to know how to retreive/calculate the exact time of each time step in the slicefile.

regards Jur

**Comment by mcgratta** [comment]
I think the times are read by slread.py
```
    while Time[st] < Tend:

        f.read(4)
        Time_list = struct.unpack('f',f.read(4))
        Time[st] = Time_list[0]
        f.read(8)
        for n in range(N):
            for m in range(M):
                QQ_list = struct.unpack('f',f.read(4))
                QQ[m,n] = QQ_list[0]
        if st%timeskip==0:
            print(st,int(st/timeskip))
            Qskip[:,:,int(st/timeskip)] = QQ[np.ix_(ii,jj)]
        f.read(4)
        st = st + 1
        if st>Nframes:
            break
```

**Comment by johodges** [comment, marked as answer]
As Kevin pointed out, the time step is part of the header information dumped with each slice. There is a subroutine in pyfdstools that will return all the timesteps in a slice file. Once you install the package, you should be able to get the information with 2 lines:

import pyfdstools as fds
times = fds.readSLCFtimes(file)

The subroutine is in this file, line 932.
https://github.com/johodges/pyfdstools/blob/master/pyfdstools/extractPlot3Ddata.py
